sound/soc/sof/pcm.c

```diff
@@ -105,19 +105,8 @@ static int sof_pcm_hw_params(struct snd_pcm_substream *substream,
 	pcm.params.host_period_bytes = params_period_bytes(params);
 
 	/* container size */
-	switch (params_width(params)) {
-	case 16:
-		pcm.params.sample_container_bytes = 2;
-		break;
-	case 24:
-		pcm.params.sample_container_bytes = 4;
-		break;
-	case 32:
-		pcm.params.sample_container_bytes = 4;
-		break;
-	default:
-		return -EINVAL;
-	}
+	pcm.params.sample_container_bytes =
+	  snd_pcm_format_physical_width(params_format(params)) >> 3;
```

Comment:
snd_pcm_format_physical_width() can return -EINVAL, so I think we need to check for that.

Reply:
yes good catch! Fixed now